components/button/button-icon.js

```diff
@@ -126,9 +128,9 @@ class ButtonIcon extends ButtonMixin(VisibleOnAncestorMixin(RtlMixin(LitElement)
 				?formnovalidate="${this.formnovalidate}"
 				formtarget="${ifDefined(this.formtarget)}"
 				name="${ifDefined(this.name)}"
-				title="${ifDefined(this.text)}"
 				type="${this._getType()}">
 				<d2l-icon icon="${ifDefined(this.icon)}" class="d2l-button-icon"></d2l-icon>
+				${this.text ? html`<d2l-tooltip for="target" disable-focus-lock>${this.text}</d2l-tooltip>` : null}
```

Comment:
Continuing conversation re API now that we're actually doing this: the current approach limits consumers to text-only for the tooltip contents.  This would not suffice for cases like GBL (which I know has it's own style applied to native button). Is Design ok with this limitation in the core components?  We could surface the slot later, but the API starts to get pretty dirty so I tend to prefer the simplicity of the text, but it is a significant difference nonetheless.  

I anticipate this approach (of baking the tooltip into the core component) to be used elsewhere, for example, potentially the card-footer-link, and I think the API should be the same.  That is, if we go text-only here, I'd expect text only there too.  The dropdown openers are another scenario where we would make use of this concept (via consuming `d2l-button-*`), and if a slot were surfaced, it would affect it's API too. 

Anyway, just to call this out/validate that limitation before we get too far along this path.

Reply:
Good to call out for sure. Since this is really just an extension of the button's label, and previously was stored in the `title` attribute, my preference would be for this to be text-only. I don't know what it would mean for rich HTML to be inside a button's label, but I'm pretty sure we don't want it.